pandas/io/stata.py

```diff
@@ -1985,24 +1978,12 @@ def _dtype_to_default_stata_fmt(dtype, column, dta_version=114, force_strl=False
         if force_strl:
             return "%9s"
     if dtype.type == np.object_:
-        inferred_dtype = infer_dtype(column, skipna=True)
-        if not (inferred_dtype in ("string", "unicode") or len(column) == 0):
-            raise ValueError(
-                "Column `{col}` cannot be exported.\n\nOnly "
```

Comment:
Is this no longer required?

Reply:
It has moved up the stack one level.  The same error message appears but it needs to happen earlier since the check happens when string data is encoded to binary for export.